bofire/surrogates/mixed_tanimoto_gp.py

```diff
@@ -0,0 +1,369 @@
+            )
+
+        transformed_X = self.inputs.transform(X, self.input_preprocessing_specs)
+
+        tX, tY = torch.from_numpy(transformed_X.values).to(**tkwargs), torch.from_numpy(
+            Y.values
+        ).to(**tkwargs)
+
+        features2idx, _ = self.inputs._get_transform_info(
+            self.input_preprocessing_specs
+        )
+
+        # List of indexes for Molecular features using Fingerprints, Fragment, FingerprintsFragments
+        mol_dims = []
+        for mol_feat in molecular_features_list:
+            for i in features2idx[mol_feat]:
+                mol_dims.append(i)
+
+        # List of indexes for Continuous inputs, Categorical inputs with descriptors, Mordred descriptors
+        ord_dims = []
+        for ord_feat in continuous_features_list:
+            for i in features2idx[ord_feat]:
+                ord_dims.append(i)
+
+        # these are the categorical dimensions after applying the OneHotToNumeric transform
+        cat_dims = list(
+            range(
+                len(ord_dims) + len(mol_dims),
+                len(ord_dims) + len(mol_dims) + len(categorical_features_list),
+            )
+        )
+
+        if len(categorical_features_list) == 0:
+            tf = scaler
+            tXX = tX
+        else:
+            # these are the categorical features within the OneHotToNumeric transform
+            categorical_features = {
+                features2idx[feat][0]: len(features2idx[feat])
+                for feat in categorical_features_list
+            }
+
+            o2n = OneHotToNumeric(
+                dim=tX.shape[1],
+                categorical_features=categorical_features,
+                transform_on_train=False,
+            )
+            tf = (
+                ChainedInputTransform(tf1=scaler, tf2=o2n)
+                if scaler is not None
+                else o2n
+            )
+            tXX = o2n.transform(tX)
+
+        # fit the model
+        self.model = MixedTanimotoGP(
+            train_X=tXX,
+            train_Y=tY,
+            cat_dims=cat_dims,
+            mol_dims=mol_dims,
+            cont_kernel_factory=partial(kernels.map, data_model=self.continuous_kernel),
+            # cat_kernel_factory=partial(kernels.map, data_model=self.categorical_kernel), BoTorch forced to use CategoricalKernel
+            mol_kernel_factory=partial(kernels.map, data_model=self.molecular_kernel),
+            outcome_transform=Standardize(m=tY.shape[-1]),
+            input_transform=tf,
+        )
+        mll = ExactMarginalLogLikelihood(self.model.likelihood, self.model)
+        fit_gpytorch_mll(mll, options=self.training_specs, max_attempts=10)
+
+    def _dumps(self) -> str:
+        """Dumps the actual model to a string via pickle as this is not directly json serializable."""
+        with warnings.catch_warnings(record=True) as w:
+            warnings.simplefilter("always")
+            import bofire.surrogates.cloudpickle_module as cloudpickle_module
```

Comment:
Why using cloudpickle here and not regular pickle?

Reply:
I have removed the cloudpickle here. It was not needed. Good catch :)